Filtered_kernel/include/CGAL/Lazy.h

```diff
@@ -1490,20 +1476,20 @@ struct Lazy_construction_variant {
     struct result<F( BOOST_PP_ENUM_PARAMS(n, T) )> {                    \
       BOOST_PP_REPEAT(n, CGAL_TYPEMAP_AC, T)                            \
       typedef typename Type_mapper<                                     \
-        typename cpp11::result_of<AC( BOOST_PP_ENUM_PARAMS(n, A) )>::type, AK, LK>::type type; \
+        decltype(std::declval<AC>()(BOOST_PP_ENUM(n, CGAL_PARAM, A))), AK, LK>::type type; \
     };
 
   BOOST_PP_REPEAT_FROM_TO(1, 9, CGAL_RESULT, _)
```

Comment:
I think the 1 to 9 was used originally to cover more possibilities in the future, which were never necessary at the end (just a guess). The `Type_mapper` from `Kernel_23` is also defined from 1 to 10. The variadic template is a great idea and indeed two operators are basically identical, however, I would not implement that in this branch since it requires to change quite a few classes and there are already a lot of changes in this branch. I would not mix them. But I could try to work on that as a separate task in the future.